### ISSUE TITLE
PR #11524 breaks packages depending on bazel directly

### ISSUE THREAD

**Issue by obreitwi** [spack/spack#11850]
PR #11524 introduced an error that breaks packages depending on bazel directly.

### Steps to reproduce the issue
Pick a random package, e.g. `htop`, and have it depend on bazel:
```console
git apply <<EOF
diff --git a/var/spack/repos/builtin/packages/htop/package.py b/var/spack/repos/builtin/packages/htop/package.py
index 63f33c466..d76748394 100644
--- a/var/spack/repos/builtin/packages/htop/package.py
+++ b/var/spack/repos/builtin/packages/htop/package.py
@@ -18,6 +18,7 @@ class Htop(AutotoolsPackage):
     version('2.0.2', '7d354d904bad591a931ad57e99fea84a')
 
     depends_on('ncurses')
+    depends_on('bazel')
 
     def configure_args(self):
         return ['--enable-shared']
EOF
$ spack install --fake htop
```

### Error Message
```console
==> Error: AttributeError: module 'spack.pkg.builtin.htop' has no attribute 'make_jobs'

spack/var/spack/repos/builtin/packages/bazel/package.py:98, in setup_dependent_package:
         95        dependent_module = inspect.getmodule(dependent_spec.package)
         96        if not dependent_spec.package.parallel:
         97            jobs = 1
  >>     98        elif dependent_module.make_jobs:
         99            jobs = dependent_module.make_jobs
        100        module.bazel = BazelExecutable('bazel', 'build', jobs)
```

### Workaround

A possible workaround is to just add `parallel = False` to the package.

**Comment by obreitwi** [comment]
Ping @alalazo as #11524 was written by him.

**Comment by obreitwi** [comment]
Ah, I forgot to answer that all `tensorflow`-related packages are affected as well…

**Comment by alalazo** [comment]
The issue here is that `dependent_module.make_jobs` is not set yet when we call the function. In my opinion the neatest solution would be to set:
```python
module.bazel = BazelExecutable('bazel', 'build')
```
and let the dependents call it with the number of jobs that seems appropriate. @obreitwi Would that be a viable solution for you? Also, in the long run, would it make sense to have a `BazelPackage` as we have a `CMakePackage`, `AutotoolsPackage`, etc. ?

**Comment by obreitwi** [comment]
> @obreitwi Would that be a viable solution for you?

It seems appropriate for me. To be honest, I found that `bazel`, more often than not, simply ignores whatever setting you set and just uses every core available its process. So as long as this bug gets resolved, I am happy (we set the number of cores available to spack via cgroups or affinity).

> Also, in the long run, would it make sense to have a `BazelPackage` as we have a `CMakePackage`, `AutotoolsPackage`, etc. ?

I think it would make a lot of sense. Every update regarding `bazel`/`tensorflow`-based packages are still "no fun", to put it mildly. Just look at how much our [`visionary`](https://github.com/electronicvisions/spack/tree/visionary)-branch still deviates from upstream/develop because we do not feel confident pushing "whatever works for us" back upstream.

Anything that streamlines the process (and thereby eases the pain) would be very welcome…!
